scheduler.py

```diff
@@ -222,6 +251,81 @@ def get_worker_round_robin_mode(self, gpuEnabledTask):
         with self.workerMapLock:
             return self.workers[pickedWorkerId]
 
+    def get_worker_power_of_two(self, gpuEnabledTask):
```

Comment:
Suggestion: For all non gpu and gpu worker combination, a recursive call can be made instead of duplicate code?

Reply:
Yes - But the method you shared does a O(n) copy of keys to a list for every call and thus maintained a separate list